apache/superset
# chore(docs): Updated recommended pip command to install PyAthena version prior to v3.0.0 release

### SUMMARY

Doesn't fix the issue, however updates the docs for: https://github.com/apache/superset/issues/23957

Advising that users lock to the PyAthena version release before the recent [v3.0.0 release](https://github.com/laughingman7743/PyAthena/releases/tag/v3.0.0) as the current recommended command now installs version 3.0.x which is causing errors

When using version <=2.x.x, errors do not occur - It appears as though there have been significant changes made to the PyAthena library.

### TESTING INSTRUCTIONS

When using version >=3.0.x of the driver, existing datasets cannot be edited, added and dashboards and charts can no longer be viewed.

```bash
2023-05-17 23:26:07 2023-05-17 15:26:07,321:ERROR:superset.views.base:name 'Union' is not defined
2023-05-17 23:26:07 Traceback (most recent call last):
2023-05-17 23:26:07   File "/app/superset/views/base.py", line 253, in wraps
2023-05-17 23:26:07     return f(self, *args, **kwargs)
2023-05-17 23:26:07   File "/app/superset/views/base_api.py", line 474, in get_headless
2023-05-17 23:26:07     duration, response = time_function(super().get_headless, pk, **kwargs)
2023-05-17 23:26:07   File "/app/superset/utils/core.py", line 1596, in time_function
2023-05-17 23:26:07     response = func(*args, **kwargs)
2023-05-17 23:26:07   File "/usr/local/lib/python3.8/site-packages/flask_appbuilder/api/__init__.py", line 1481, in get_headless
2023-05-17 23:26:07     response[API_RESULT_RES_KEY] = show_model_schema.dump(item, many=False)
2023-05-17 23:26:07   File "/usr/local/lib/python3.8/site-packages/marshmallow/schema.py", line 552, in dump
2023-05-17 23:26:07     result = self._serialize(processed_obj, many=many)
2023-05-17 23:26:07   File "/usr/local/lib/python3.8/site-packages/marshmallow/schema.py", line 520, in _serialize
2023-05-17 23:26:07     value = field_obj.serialize(attr_name, obj, accessor=self.get_attribute)
2023-05-17 23:26:07   File "/usr/local/lib/python3.8/site-packages/marshmallow/fields.py", line 330, in serialize
2023-05-17 23:26:07     value = self.get_value(obj, attr, accessor=accessor)
2023-05-17 23:26:07   File "/usr/local/lib/python3.8/site-packages/marshmallow/fields.py", line 259, in get_value
2023-05-17 23:26:07     return accessor_func(obj, check_key, default)
2023-05-17 23:26:07   File "/usr/local/lib/python3.8/site-packages/marshmallow/schema.py", line 477, in get_attribute
2023-05-17 23:26:07     return get_value(obj, attr, default)
2023-05-17 23:26:07   File "/usr/local/lib/python3.8/site-packages/marshmallow/utils.py", line 239, in get_value
2023-05-17 23:26:07     return _get_value_for_key(obj, key, default)
2023-05-17 23:26:07   File "/usr/local/lib/python3.8/site-packages/marshmallow/utils.py", line 253, in _get_value_for_key
2023-05-17 23:26:07     return getattr(obj, key, default)
2023-05-17 23:26:07   File "/app/superset/connectors/sqla/models.py", line 779, in select_star
2023-05-17 23:26:07     return self.database.select_star(
2023-05-17 23:26:07   File "/app/superset/models/core.py", line 635, in select_star
2023-05-17 23:26:07     eng = self._get_sqla_engine(schema=schema, source=utils.QuerySource.SQL_LAB)
2023-05-17 23:26:07   File "/app/superset/models/core.py", line 503, in _get_sqla_engine
2023-05-17 23:26:07     raise self.db_engine_spec.get_dbapi_mapped_exception(ex)
2023-05-17 23:26:07   File "/app/superset/models/core.py", line 501, in _get_sqla_engine
2023-05-17 23:26:07     return create_engine(sqlalchemy_url, **params)
2023-05-17 23:26:07   File "<string>", line 2, in create_engine
2023-05-17 23:26:07   File "/usr/local/lib/python3.8/site-packages/sqlalchemy/util/deprecations.py", line 309, in warned
2023-05-17 23:26:07     return fn(*args, **kwargs)
2023-05-17 23:26:07   File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/create.py", line 534, in create_engine
2023-05-17 23:26:07     entrypoint = u._get_entrypoint()
2023-05-17 23:26:07   File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/url.py", line 661, in _get_entrypoint
2023-05-17 23:26:07     cls = registry.load(name)
2023-05-17 23:26:07   File "/usr/local/lib/python3.8/site-packages/sqlalchemy/util/langhelpers.py", line 330, in load
2023-05-17 23:26:07     return self.impls[name]()
2023-05-17 23:26:07   File "/usr/local/lib/python3.8/site-packages/importlib_metadata/__init__.py", line 210, in load
2023-05-17 23:26:07     module = import_module(match.group('module'))
2023-05-17 23:26:07   File "/usr/local/lib/python3.8/importlib/__init__.py", line 127, in import_module
2023-05-17 23:26:07     return _bootstrap._gcd_import(name[level:], package, level)
2023-05-17 23:26:07   File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
2023-05-17 23:26:07   File "<frozen importlib._bootstrap>", line 991, in _find_and_load
2023-05-17 23:26:07   File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
2023-05-17 23:26:07   File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
2023-05-17 23:26:07   File "<frozen importlib._bootstrap_external>", line 843, in exec_module
2023-05-17 23:26:07   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
2023-05-17 23:26:07   File "/app/superset_home/.local/lib/python3.8/site-packages/pyathena/sqlalchemy/rest.py", line 2, in <module>
2023-05-17 23:26:07     from pyathena.sqlalchemy.base import AthenaDialect
2023-05-17 23:26:07   File "/app/superset_home/.local/lib/python3.8/site-packages/pyathena/sqlalchemy/base.py", line 861, in <module>
2023-05-17 23:26:07     class AthenaDialect(DefaultDialect):
2023-05-17 23:26:07   File "/app/superset_home/.local/lib/python3.8/site-packages/pyathena/sqlalchemy/base.py", line 979, in AthenaDialect
2023-05-17 23:26:07     def _get_table(self, connection, table_name: str, schema: Optional[str] = None, **kw):
2023-05-17 23:26:07   File "/usr/local/lib/python3.8/site-packages/sqlalchemy/util/langhelpers.py", line 198, in decorate
2023-05-17 23:26:07     decorated = _exec_code_in_env(code, env, fn.__name__)
2023-05-17 23:26:07   File "/usr/local/lib/python3.8/site-packages/sqlalchemy/util/langhelpers.py", line 228, in _exec_code_in_env
2023-05-17 23:26:07     exec(code, env)
2023-05-17 23:26:07   File "<string>", line 1, in <module>
```

![image](https://github.com/apache/superset/assets/7262516/de58bb4d-6d56-4c48-a3d3-f3a1c4faa522)

![image](https://github.com/apache/superset/assets/7262516/4061491c-d2df-4f2e-a27d-529eef2a1320)

### ADDITIONAL INFORMATION
